aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/runtime_checks.move

```diff
@@ -0,0 +1,187 @@
+/// Aggregators and other structs that use delayed fields have certain
+/// restrictions imposed by runtime, e.g. they cannot be compared, serialized,
+/// etc.
+module 0x1::runtime_checks {
+    use std::bcs;
+    use std::string;
+    use aptos_std::string_utils;
+    use aptos_framework::aggregator_v2::{Self, Aggregator, AggregatorSnapshot, DerivedStringSnapshot};
+
+    //
+    // Structs and constructors.
+    //
+
+    struct StructWithAggregator<IntTy: copy + drop> has drop {
+        aggregator: Aggregator<IntTy>,
+    }
+
+    struct StructWithAggregatorSnapshot<IntTy: copy + drop> has drop {
+        snapshot: AggregatorSnapshot<IntTy>,
+    }
+
+    struct StructWithDerivedStringSnapshot has drop {
+        derived_string_snapshot: DerivedStringSnapshot,
+    }
+
+    fun with_aggregator<IntTy: copy + drop>(): StructWithAggregator<IntTy> {
+        StructWithAggregator {
+            aggregator: aggregator_v2::create_unbounded_aggregator<IntTy>()
+        }
+    }
+
+    fun with_snapshot<IntTy: copy + drop>(value: IntTy): StructWithAggregatorSnapshot<IntTy> {
+        StructWithAggregatorSnapshot {
+            snapshot: aggregator_v2::create_snapshot(value)
+        }
+    }
+
+    fun with_derived_string_snapshot(str: vector<u8>): StructWithDerivedStringSnapshot {
+        StructWithDerivedStringSnapshot {
+            derived_string_snapshot: aggregator_v2::create_derived_string(string::utf8(str))
+        }
+    }
+
+    //
+    // Equality.
+    //
+
+    public entry fun test_equality_with_aggregators_I() {
+        let a = with_aggregator<u64>();
+        let b = with_aggregator<u64>();
+        aggregator_v2::try_add(&mut b.aggregator, 10);
+        assert!(a != b, 0);
+    }
+
+    public entry fun test_equality_with_aggregators_II() {
+        let a = with_aggregator<u64>();
+        let b = with_aggregator<u64>();
+        assert!(a == b, 0);
```

Comment:
These tests look simple. Do they need to be in e2e tests? Seems like these assertions could be done in unit tests as well?

Reply:
As we can't compare native values, shouldn't these operations return error? How come these operations succeed?

Reply:
1. I think it is because of feature flags, aggregators are not enabled so regular move tests do not work. But we enabled the flag in genesis recenlty, I will rebase and try to move them in this case. Thanks for pointing out!
2. `assert!(a == b, 0);` is never called, because `==` fails. I think this is confusing, so I changed the tests to 
```
let a = with_aggregator<u64>();
let b = with_aggregator<u64>();
let _ = a == b;
```
to avoid confusion.